src/libs/ui/mainwindow.cpp

```diff
@@ -568,7 +585,12 @@ void MainWindow::syncTreeView()
         oldSelectionModel->deleteLater();
     }
 
+    // Connect to new selection model
+    connect(ui->treeView->selectionModel(), &QItemSelectionModel::selectionChanged,
```

Comment:
Connection should be done only for the new selection model. Plus there's really no need in the `docsetSelectionChanged` slot. A lambda will do just fine.
I suggest rewriting this the following way:

```cpp
    // TODO: Remove once QTBUG-49966 is addressed.
    QItemSelectionModel *newSelectionModel = ui->treeView->selectionModel();
    if (newSelectionModel != oldSelectionModel) {
        if (oldSelectionModel) {
            oldSelectionModel->deleteLater();
        }

        connect(newSelectionModel, &QItemSelectionModel::selectionChanged,
                this, [this](const QItemSelection &selected) {
            if (selected.isEmpty()) {
                return;
            }

            m_openDocsetTimer->start();
            m_openDocsetTimer->setProperty("index", selected.indexes().first());
        });
    }
```

Reply:
I'll still inline this one though